File: cpp_example/create_addressbook.cpp
#include "addressbook.pb.h"
#include <fstream>
#include <iostream>

void createAddressBook(){
  tutorial::AddressBook addressbook;

  tutorial::Person* person1 = addressbook.add_people();
  person1->set_name("John");
  person1->set_id(123);
  person1->set_email("john@example.com");

  tutorial::Person::PhoneNumber* phone1 = person1->add_phones();
  phone1->set_number("123456789");
  phone1->set_type(tutorial::Person_PhoneType::Person_PhoneType_PHONE_TYPE_MOBILE);

  tutorial::Person* person2 = addressbook.add_people();
  person2->set_name("Jane");
  person2->set_id(456);
  person2->set_email("jane@example.com");

  tutorial::Person::PhoneNumber* phone2 = person2->add_phones();
  phone2->set_number("987654321");
  phone2->set_type(tutorial::Person_PhoneType::Person_PhoneType_PHONE_TYPE_HOME);

  std::ofstream output("./addressbook.pb", std::ios::binary);
  if(addressbook.SerializeToOstream(&output)){
    std::cout<<"addressbook write file success"<<std::endl;
  }
  else{
    std::cout<<"addressbook write file failed"<<std::endl;
  }
  output.close();
}

int main(){
  // createAddressBook();
  std::cout<<"hello world"<<std::endl;
  return 0;
}

File: python_example/create_addressbook.py
import addressbook_pb2

def create_address_book():
    # 创建一个 AddressBook 实例
    address_book = addressbook_pb2.AddressBook()

    # 创建第一个 Person 实例
    person1 = address_book.people.add()
    person1.name = "John"
    person1.id = 123
    person1.email = "john@example.com"

    # 添加第一个 Person 的电话号码
    phone1 = person1.phones.add()
    phone1.number = "1234567890"
    phone1.type = addressbook_pb2.Person.PHONE_TYPE_MOBILE

    # 创建第二个 Person 实例
    person2 = address_book.people.add()
    person2.name = "Jane"
    person2.id = 456
    person2.email = "jane@example.com"

    # 添加第二个 Person 的电话号码
    phone2 = person2.phones.add()
    phone2.number = "9876543210"
    phone2.type = addressbook_pb2.Person.PHONE_TYPE_HOME

    return address_book

# 创建 AddressBook 实例
address_book = create_address_book()

# 将 AddressBook 序列化为字节串
serialized_data = address_book.SerializeToString()

# 将 AddressBook 写入文件
with open("./addressbook.pb", "wb") as f:
    f.write(serialized_data)
